### PR TITLE
Kozmaz87/small fixes

### LINKED ISSUE #242: Add http proxy support
Which I did just move it to a branch and merge it.

### LINKED ISSUE #243: Segmentation discard functionality. Make sure it works.
It is leaking memory for some reason. Investigate

### PR DESCRIPTION
**Pull request by kozmaz87**
This pull request contains minor ports of some code that was written for the demo branch and also adds testing for memory leaks adds discard functionality to the consumer classes.
closes #242 
closes #243

### CODE REVIEW

File: ebu_tt_live/scripts/ebu_ebuttd_encoder.py [hunk 1]
```diff
@@ -39,6 +39,8 @@
                     default='current', metavar='HH:MM:SS.mmm')
 parser.add_argument('-o', '--output-folder', dest='output_folder', default='./')
 parser.add_argument('-of', '--output-format', dest='output_format', default='xml')
+parser.add_argument('--proxy', dest='proxy', help='Proxy server', type=str, metavar='PROXY:PORT')
```
Comment (nigelmegitt) [inline ebu_tt_live/scripts/ebu_ebuttd_encoder.py]: Would be useful to clarify if the protocol prefix is expected here or not, e.g. should it be `http://server_path:port` or just `server_path::port` ?

File: ebu_tt_live/scripts/ebu_simple_consumer.py [hunk 2]
```diff
@@ -30,6 +30,7 @@
                     help='Works only with -m, if set the script will wait for new lines to be added to the file once the last line is reached. Exactly like tail -f does.',
                     action="store_true", default=False
                     )
+parser.add_argument('--proxy', dest='proxy', help='Proxy server', type=str, metavar='PROXY:PORT')
```
Comment (nigelmegitt) [inline ebu_tt_live/scripts/ebu_simple_consumer.py]: As above, useful to describe if protocol is needed, in the help string.

File: ebu_tt_live/scripts/test/test_scripts.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+
+from subprocess import Popen, PIPE
+from unittest import TestCase
+from ebu_tt_live.scripts import ebu_dummy_encoder
+
+
+class TestDummyScript(TestCase):
+
+    def test_simple_run(self):
+        process = Popen('ebu-dummy-encoder', stderr=PIPE, stdout=PIPE)
```
Comment (nigelmegitt) [inline ebu_tt_live/scripts/test/test_scripts.py]: How does this work if the system firewall needs confirmation that the process should accept incoming connections?

Comment (kozmaz87) [inline ebu_tt_live/scripts/test/test_scripts.py]: This is exactly the reason it is only the dummy encoder that is in this file for the moment. Twisted supports mocking out networking via their twisted.trial package and make it work. I will come to do it when we get a bit more time. #244 stays open for this reason.

Comment (nigelmegitt) [inline ebu_tt_live/scripts/test/test_scripts.py]: Ah right, interesting.

**Comment by nigelmegitt** [comment]
Looks like a summary of the command line parameters could be added to `docs/source/overview.rst` but is not essential here, so I'm happy with this as is.

**Pull request merged**